rpc/src/v1/helpers/signature.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethkey::{recover, public_to_address, Signature};
+use jsonrpc_core::Result;
+use v1::types::{Bytes, BasicAccount, H256, U64};
+use v1::helpers::errors;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+
+/// helper method for parity_verifySignature
+pub fn verify_signature(
+	is_prefixed: bool,
+	message: Bytes,
+	r: H256,
+	s: H256,
+	v: U64,
+	chain_id: Option<u64>
+) -> Result<BasicAccount> {
+	let hash = if is_prefixed {
+		eth_data_hash(message.0)
+	} else {
+		keccak(message.0)
+	};
+	let v: u64 = v.into();
+	let is_valid_for_current_chain = match (chain_id, v) {
+		(None, v) if v == 0 || v == 1 => true,
+		(Some(chain_id), v) if v > 36 => (v - 35) / 2 == chain_id,
+		_ => false,
+	};
+
+	let v = if v > 36 { (v - 1) % 2 } else { v };
+
+	let signature = Signature::from_rsv(&r.into(), &s.into(), v as u8);
+	let public = recover(&signature, &hash).map_err(errors::encryption)?;
+	let address = public_to_address(&public);
+	Ok(BasicAccount { address, public_key: public, is_valid_for_current_chain })
```

Comment:
hrm, if `is_valid_for_current_chain == false` isn't better to return Error here? Or please explain :1st_place_medal:

Reply:
no it's included in the return payload.